src/main/resources/public/js/rooms.js

```diff
@@ -132,3 +152,17 @@ function updateTable() {
         table.columns.adjust().draw();
     });
 }
+
+$.fn.dataTable.ext.search.push(
+    function( settings, data, dataIndex ) {
+        var status = data[0].toLowerCase();
+        var type = data[3];
+        var price = data[5];
+        var size = data[4];
+
+        if ($('#' + status).is(":checked") && $('#' + type).is(":checked") && $('#size-' + size).is(":checked")) {
```

Comment:
could rewrite the if statement as return statement as it returns true or false anyway